message_adapter/message_adapter.py

```diff
@@ -52,9 +52,10 @@ def __getTaskNameFromExecutionHistory(self, executionHistory, arn):
             # previousEventId's (TaskStateEntered) name
             if (arn is not None and
                     ((step['type'] == 'LambdaFunctionScheduled' and
-                        step['lambdaFunctionScheduledEventDetails']['resource'] == arn) or
-                        (step['type'] == 'ActivityScheduled' and
-                            step['activityScheduledEventDetails']['resource'] == arn))):
+                      step['lambdaFunctionScheduledEventDetails']['resource'] == arn) or
+                     (step['type'] == 'ActivityScheduled' and
+                      step['activityScheduledEventDetails']['resource'] == arn)) and
+                    'stateEnteredEventDetails' in eventsById[step['previousEventId']]):
```

Comment:
In case of step['type'] == 'ActivityScheduled' and retry, how does this method return the correct task name then?

Reply:
In that case, `step['previousEventId']` should be of type `ActivityFailed`. In this case, CuMA will step through the events list (from most recent to least recent) until it hits the `ActivityScheduled` such that its previous step has the `stateEnteredEventDetails` attribute (i.e the step immediately before which the task state was entered.

The issue that this is attempting to fix is that, in the event that we're in a retry, `step['previousEventId']` could be of type `ActivityFailed`, which doesn't have a `stateEnteredEventDetails` attribute. This causes a KeyError to be thrown.

Reply:
Thanks @ifestus.  Looks good to me.